src/libclient/canvas/paintengine.cpp

```diff
@@ -139,21 +141,26 @@ void PaintEngine::enqueueReset()
 	receiveMessages(false, 1, &msg);
 }
 
-void PaintEngine::enqueueLoadBlank(const QSize &size, const QColor &backgroundColor)
+void PaintEngine::enqueueLoadBlank(
+	int undoDepthLimit, const QSize &size, const QColor &backgroundColor)
 {
 	drawdance::Message messages[] = {
 		drawdance::Message::makeInternalReset(0),
+		drawdance::Message::makeUndoDepth(0, undoDepthLimit),
```

Comment:
This is almost certainly irrelevant now but figured I would mention as an idea for the future that not having atomicity in the canvas reset messages was an annoyance when I was trying to keep the canvas view stable across auto-resets. This was probably more the fault of the convoluted path messages take to that made it hard to figure out how to hook it properly, but this feels overnormalised.

Reply:
Yup, I agree. Non-atomic resets still cause problems, like messing up the layer list and such.